### ISSUE TITLE
False positive in `prefer-array-flat`

### ISSUE THREAD

**Issue by omril1** [sindresorhus/eslint-plugin-unicorn#1707]
A false positive with some possibilities for other improvements/rules.

The rule matches the first concat during consecutive calls with an empty array

```js
const foo = [].concat(arrayA).concat(arrayB);
//          ~~~~~~~~~~~~~~~~~
```
Applying the fixer results in code that doesn't make sense
```js
const foo = [arrayA].flat().concat(arrayB);
```

`[arrayA].flat()` can be simplified to just `arrayA` (maybe need a rule for it?) and the intention for the consecutive calls which I found in some legacy code probably meant to combine the tow arrays into a new one (which could probably be done similarly to the `no-array-push-push` rule, so maybe add a `no-array-concat-concat` rule?)

**Comment by fisker** [comment]
```js
const  arrayA = 1;
const foo = [].concat(arrayA).concat(arrayB);
```
